### ISSUE TITLE
Definitions and explanations for “Set of Documents” and “Set of Software Programs” produces strange corner cases that should be addressed or explained

### ISSUE THREAD

**Issue by NoahGresser** [w3c/wcag2ict#473]
The draft reads:

> The term **set of documents**, as used in WCAG2ICT, has the meaning below:
> 
> **set of documents (non-web) (as used in WCAG2ICT)**
> collection of **[[documents](https://w3c.github.io/wcag2ict/#document)]** that share a common purpose; are created by the same author, group or organization; **[are published together; and all refer to each other by name or link]**

and

> The term **set of software programs**, as used in WCAG2ICT, has the meaning below:
> 
> **set of software programs (as used in WCAG2ICT)**
> collection of **[[software](https://w3c.github.io/wcag2ict/#software) programs]** that share a common purpose; are created by the same author, group or organization; **[and are distributed together and can be launched and used independently from each other, but are interlinked each with every other one such that users can navigate from one program to another via a consistent method that appears in each member of the set]**

The way these definitions are worded, it seems that if there was a group of 5 documents/software programs (for example, let’s say documents A, B, C, D, and E), if documents A, B, and C were each linked to all of the other documents, but documents D and E were not linked to each other, then the group of documents "A, B, C, D, and E" cannot be categorized as a “Set of Documents”, even though the groups of document "A, B, C, and D" and "A, B, C, and E" can. This seems like it could create strange corner cases where an evaluation would have to identify multiple smaller sets of documents/software programs, instead of the larger group that technically does not meet the definition, in order for the criteria that refer to a “Set of Documents”/“Set of Software Programs” to apply.

I feel these definitions should either be amended to remove the corner cases, or a note should be added explicitly identifying them and explaining how to potentially handle them.

**Comment by GreggVan** [comment]
they were specifically crafted that way -- to exclude things that were not part of the set.   

as for your example -- think about it.     You end up with exactly the same result for consistency -- that you want to  but not for navigation -- because you don't.   So it works.

**Comment by NoahGresser** [comment]
My point was that it does not really make sense to me that those items are not considered part of the set. The requirement that they "refer to each other by name or link" or "are interlinked each with every other one" is not a translation of existing wording of the "set of web pages" definition, they are entirely new restrictions. While I understand the necessity to have some form of restriction, I feel the current definitions exclude things that should not be excluded. So if the definition is not going to be changed, I think there should at least be an explicit mention/explanation of what I expect to be a realtively common corner case.
Regarding the following:
> as for your example -- think about it. You end up with exactly the same result for consistency -- that you want to but not for navigation -- because you don't. So it works.

I am not entirely sure I understand what you are trying to say. From what I can gather, you are trying to say that the criteria which refer to “set of non-web documents” or “set of software programs” (e.g. 2.4.1 Bypass Blocks, 2.4.5 Multiple Ways, 3.2.3 Consistent Navigation, 3.2.4 Consistent Identification, and 3.2.6 Consistent Help) still apply because each document is part of a set. So even though documents D and E are not part of the same set, the overlap in their respective sets (A, B, and C) effectively cause the criteria to apply due to a sort of transitive relation.

If my interpretation is correct, then that is something I think should be explicitly explained in a note, because that was not obvious to me from the current definition.

**Comment by maryjom** [comment]
@NoahGresser Thank you for your question regarding WCAG2ICT. The Task Force considered making the advisory content regarding “sets of software” and “sets of documents” more formal, but decided against making any changes. While overlapping sets could occur, we do not foresee a problem in evaluating each overlapping set as a set.
